### PR TITLE
adding name as TranslatableString

### PR DESCRIPTION
**Pull request by wiewiurdp**

### CODE REVIEW

File: module/category-tree/src/Application/Controller/Api/CategoryTreeController.php [hunk 1]
```diff
@@ -203,8 +196,9 @@ public function createTree(Request $request): Response
 
         if ($name && $code) {
```
Comment (BastekBielawski) [inline module/category-tree/src/Application/Controller/Api/CategoryTreeController.php]: Create a standard form for this

File: module/category-tree/src/Domain/Entity/CategoryTree.php [hunk 2]
```diff
@@ -59,11 +71,29 @@ public function getId(): AbstractId
...
+     */
+    public function getName(): TranslatableString
     {
         return $this->name;
     }
 
+    /**
+     * @param TranslatableString $title
+     */
+    public function changeTitle(TranslatableString $title): void
```
Comment (BastekBielawski) [inline module/category-tree/src/Domain/Entity/CategoryTree.php]: What title? We have name and code

File: module/category-tree/src/Domain/Event/CategoryTreeNameChangedEvent.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+namespace Ergonode\CategoryTree\Domain\Event;
+
+use Ergonode\Core\Domain\ValueObject\TranslatableString;
+use Ergonode\EventSourcing\Infrastructure\DomainEventInterface;
+use JMS\Serializer\Annotation as JMS;
+
+/**
+ */
+class CategoryTreeNameChangedEvent implements DomainEventInterface
```
Comment (BastekBielawski) [inline module/category-tree/src/Domain/Event/CategoryTreeNameChangedEvent.php]: I think we have AbstractTranslatableChangedEvent?

File: module/category-tree/src/Persistence/Dbal/Projector/CategoryTreeNameChangedEventProjector.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+use Ergonode\EventSourcing\Infrastructure\DomainEventInterface;
+use Ergonode\EventSourcing\Infrastructure\Exception\ProjectorException;
+use Ergonode\EventSourcing\Infrastructure\Exception\UnsupportedEventException;
+use Ergonode\EventSourcing\Infrastructure\Projector\DomainEventProjectorInterface;
+
+/**
+ */
+class CategoryTreeNameChangedEventProjector implements DomainEventProjectorInterface
+{
+
```
Comment (BastekBielawski) [inline module/category-tree/src/Persistence/Dbal/Projector/CategoryTreeNameChangedEventProjector.php]: Unnecessery empty line

File: module/category-tree/src/Domain/Updater/CategoryTreeUpdater.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+declare(strict_types = 1);
+
+namespace Ergonode\CategoryTree\Domain\Updater;
+
+use Ergonode\CategoryTree\Domain\Entity\CategoryTree;
+use Ergonode\Core\Domain\ValueObject\TranslatableString;
+
+/**
+ */
+class CategoryTreeUpdater
```
Comment (rprzedzik) [inline module/category-tree/src/Domain/Updater/CategoryTreeUpdater.php]: This class is not necessary you can move it to TreeUpdateHandler

File: module/category-tree/src/Infrastructure/Grid/TreeGrid.php [hunk 6]
```diff
@@ -46,7 +44,11 @@ public function init(GridConfigurationInterface $configuration, Language $langua
         $id = new TextColumn('id', 'Id');
         $id->setVisible(false);
         $this->addColumn('id', $id);
-        $name = new TextColumn('name', 'Name', new TextFilter($filters->getString('name')));
+        $code = new TextColumn('code', 'Code', new TextFilter($filters->getString('code')));
+        $code->setWidth(280);
+        $this->addColumn('code', $code);
+        $this->orderBy('code', 'ASC');
+        $name = new TextColumn('name', 'Code', new TextFilter($filters->getString('name')));
```
Comment (rprzedzik) [inline module/category-tree/src/Infrastructure/Grid/TreeGrid.php]: Change column name to "Name"

File: symfony.lock [hunk 7]
```diff
@@ -515,9 +515,6 @@
             "ref": "85ab0b55d90a8f3e29e1b685a71f5b37923a7f5e"
         }
     },
-    "symfony/mime": {
```
Comment (rprzedzik) [inline symfony.lock]: Why you add chanes to symofny lock file ???

File: module/category-tree/src/Domain/Event/CategoryTreeNameChangedEvent.php (hunk 3 above)
Comment (wiewiurdp) [reply module/category-tree/src/Domain/Event/CategoryTreeNameChangedEvent.php]: I couldn't find it

File: module/category-tree/src/Domain/Entity/CategoryTree.php [hunk 8]
```diff
@@ -59,11 +71,29 @@ public function getId(): AbstractId
...
+     */
+    public function getName(): TranslatableString
     {
         return $this->name;
     }
 
+    /**
+     * @param TranslatableString $title
+     */
+    public function changeName(TranslatableString $title): void
```
Comment (BastekBielawski) [inline module/category-tree/src/Domain/Entity/CategoryTree.php]: Change to ``$name``

**Pull request merged**